engine/gamesys/src/gamesys/scripts/script_buffer.cpp

```diff
@@ -879,6 +880,239 @@ namespace dmGameSystem
+     *
+     * @name buffer.get_metadata
+     * @param buf [type:buffer] the buffer to get the metadata from
+     * @param metadata_name [type:hash|string] name of the metadata entry
+     * @return values [type:table] table of metadata values or nil if the entry does not exist
+     * @return value_type [type:constant] numeric type of values or nil
+     *
+     * @examples
+     * How to get a metadata entry from a buffer
+     *
+     * ```lua
+     * -- retrieve a metadata entry named "somefloats" and its nomeric type
+     * local values, type = buffer.get_metadata(buf, hash("somefloats"))
+     * if metadata then print(#metadata.." values in 'somefloats'") end
+     * ```
+    */
+    static int GetMetadata(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 2);
+
+        // get buffer
+        dmScript::LuaHBuffer* buffer = dmScript::CheckBuffer(L, 1);
+        dmBuffer::HBuffer hbuffer = UnpackLuaBuffer(buffer);
+
+        // get metadata entry name
+        dmhash_t entry_name = dmScript::CheckHashOrString(L, 2);
+
+        uint32_t count;
+        dmBuffer::ValueType valueType;
+        void* values;
+        dmBuffer::Result r = dmBuffer::GetMetaData(hbuffer, entry_name, &values, &count, &valueType);
+        if ( r == dmBuffer::RESULT_METADATA_MISSING)
+        {
+            lua_pushnil(L);  // nil for metadata entry
+            lua_pushnil(L); // nil for numbertype
+            return 2;
+        }
+        if ( r != dmBuffer::RESULT_OK )
+        {
+            return DM_LUA_ERROR("%s.%s: error getting metadata for buffer: %s", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA, dmBuffer::GetResultString(r));
+        }
+
+        lua_newtable(L);
+        #define DM_ARRAY_TO_LUA_TABLE(_T_) ArrayToLuaTable<_T_>(L, (_T_*) values, count, valueType)
+        switch (valueType)
+        {
+            case dmBuffer::VALUE_TYPE_FLOAT32:
+                DM_ARRAY_TO_LUA_TABLE(float);
+            break;
+            case dmBuffer::VALUE_TYPE_UINT8:
+                DM_ARRAY_TO_LUA_TABLE(uint8_t);
+            break;
+            case dmBuffer::VALUE_TYPE_UINT16:
+                DM_ARRAY_TO_LUA_TABLE(uint16_t);
+            break;
+            case dmBuffer::VALUE_TYPE_UINT32:
+                DM_ARRAY_TO_LUA_TABLE(uint32_t);
+            break;
+            case dmBuffer::VALUE_TYPE_INT8:
+                DM_ARRAY_TO_LUA_TABLE(int8_t);
+            break;
+            case dmBuffer::VALUE_TYPE_INT16:
+                DM_ARRAY_TO_LUA_TABLE(int16_t);
+            break;
+            case dmBuffer::VALUE_TYPE_INT32:
+                DM_ARRAY_TO_LUA_TABLE(int32_t);
+            break;
+            case dmBuffer::VALUE_TYPE_UINT64:
+            case dmBuffer::VALUE_TYPE_INT64:
+                return DM_LUA_ERROR("%s.%s retrieving 64 bit integer types is not supported.", SCRIPT_LIB_NAME, SCRIPT_TYPE_NAME_BUFFERMETADATA);
+                return DM_LUA_ERROR("invalid value type supplied: %d.", valueType);
```

Comment:
Double return statements.
I'm assuming the compiler warned about this?

Reply:
oh,  it seems I introduced that in a last time fix. 
It (the compiler) probably did :-)